onnx_tf/backends/backend_v1.py

```diff
@@ -93,25 +92,43 @@ def _get_output_shape(auto_pad, input_spatial_shape, kernel_spatial_shape,
                   / float(strides_spatial[i])))
       return out_shape
 
+    spatial_size = len(x_shape)
+    new_pads = pads[:]
     if auto_pad in ["SAME_UPPER", "SAME_LOWER"]:
-      out_shape = _get_output_shape(auto_pad, x_shape[2:], kernel_shape,
-                                    strides)
-      pad_shape = _get_pad_shape(auto_pad, x_shape[2:], kernel_shape, strides,
+      out_shape = _get_output_shape(auto_pad, x_shape, kernel_shape, strides)
+      pad_shape = _get_pad_shape(auto_pad, x_shape, kernel_shape, strides,
                                  out_shape)
       for i in range(spatial_size):
         if auto_pad == "SAME_LOWER":
-          pads[i + spatial_size] = pad_shape[i] // 2
-          pads[i] = pad_shape[i] - pads[i + spatial_size]
+          new_pads[i + spatial_size] = pad_shape[i] // 2
+          new_pads[i] = pad_shape[i] - new_pads[i + spatial_size]
         elif auto_pad == "SAME_UPPER":
-          pads[i] = pad_shape[i] // 2
-          pads[i + spatial_size] = pad_shape[i] - pads[i]
+          new_pads[i] = pad_shape[i] // 2
+          new_pads[i + spatial_size] = pad_shape[i] - new_pads[i]
     elif auto_pad in ["", "VALID"]:
       pad_shape = [
           pads[i] + pads[i + spatial_size] for i in range(spatial_size)
       ]
-      out_shape = _get_output_shape(auto_pad, np.add(x_shape[2:], pad_shape),
+      out_shape = _get_output_shape(auto_pad, np.add(x_shape, pad_shape),
                                     kernel_shape, strides)
-    return out_shape, pad_shape, pads
+    return out_shape, new_pads
+
+  # input_shape, kernel_shape, strides are specified for
+  # spatial dims only.
+  @classmethod
+  def get_tf_pad(cls, input_shape, kernel_shape, strides, pads):
+    assert pads is not None
+    num_sp_dim = int(len(kernel_shape))
+
+    if pads == [0] * num_sp_dim * 2 or pads is None:
```

Comment:
I had `assert pads is not None` right above...

Reply:
..just copy paste the old one.....